examples/mir_demo_server/server_example.cpp

```diff
@@ -117,7 +118,272 @@ catch (...)
+                                                    else
+                                                    {
+                                                        mir::log_warning(
+                                                            "Config key '%s' has invalid value: %s",
+                                                            key.to_string().c_str(),
+                                                            val->data());
+                                                    }
+                                                }
+                                                filter(new_filter);
+                                            },
+                                            {"output_filter"});
+    }
+};
+
+// Struct for illustrative purposes, this would be rolled into miral::InputConfiguration
+struct InputConfiguration : miral::InputConfiguration
+{
+    struct State
+    {
+        State(miral::InputConfiguration::Mouse mouse,
+        miral::InputConfiguration::Touchpad touchpad,
+        miral::InputConfiguration::Keyboard keyboard) :
+        mouse{mouse},
+        touchpad{touchpad},
+        keyboard{keyboard}
+        {}
+
+        std::mutex mutex;
+        miral::InputConfiguration::Mouse mouse;
+        miral::InputConfiguration::Touchpad touchpad;
+        miral::InputConfiguration::Keyboard keyboard;
+    };
+
+    std::shared_ptr<State> const state = std::make_shared<State>(mouse(), touchpad(), keyboard());
+
+    explicit InputConfiguration(AttributeHandler& config_handler) : miral::InputConfiguration{}
+    {
+        config_handler.add_string_attribute([this](ConfigurationKey const& key, std::optional<std::string_view> val)
+                                            {
+                                                if (val)
+                                                {
+                                                    std::lock_guard lock{state->mutex};
+                                                    auto const& value = *val;
+                                                    if (value == "right")
+                                                        state->mouse.handedness(mir_pointer_handedness_right);
+                                                    else if (value == "left")
+                                                        state->mouse.handedness(mir_pointer_handedness_left);
+                                                    else
+                                                        mir::log_warning(
+                                                            "Config key '%s' has invalid value: %s",
+                                                            key.to_string().c_str(),
+                                                            val->data());
+                                                }
+                                            },
+                                            {"pointer", "handedness"});
```

Comment:
Would these also be under `input` as a general category? Just asking as I would do this in miracle regardless

```suggestion
                                            {"input", "pointer", "handedness"});
```

Reply:
As discussed above I was thinking more along the lines of:

```c++
    InputConfiguration input_configuration{mattkae_configuration.node("input")};
```

But including the "namespace" in the key is certainly an alternative